### ISSUE TITLE
[crash loop] After botched file transfer, can't reopen Conversations

### ISSUE THREAD

**Issue by axelsimon** [iNPUTmice/Conversations#1516]
TL;DR:
- Conversation crashed on picture transfer
- Restarted Conversations, sent bug report
- Conversations slowed to a grinding halt, got killed by Android
- Conversations won't start again, local message DBs are likely crashed

Hi,
A strange thing happened yesterday.
My correspondent sent me a picture, which was received and crashed Conversations. This has happened often in the past and is not a big deal: just starting Conversations again takes me straight back to the conversation I was having, the picture is there, and I just keep going. On starting Conversations I get a request to send the bug report, which I agreed to in this time.

However, in this case, restarting Conversations didn't work out so well: right after I sent the bugreport, Conversations became so slow it was unusable. Other apps seemed ok, but every time I used the activity switcher to go back to Conversations, it was still unresponsive.
After a while, Android killed Conversations. I started it again, and it just crashed.
After a few tries, I rebooted, and same problem, starting Conversations just crashes it.

From a quick loot at the logs, my understanding is that the local database in which Conversation stores its messages got corrupted/unreadable in some way, and Conversations crashes trying to read from it/them.

I will try to provide a logcat to support this.

Thanks

axel

**Comment by axelsimon** [comment]
If I'm not mistaken, here is the relevant part of the log:

````
D/conversations( 4257): AxolotlService (account2@example.net): Building session for remote address: account1@example.net:1499453371
D/conversations( 4257): AxolotlService (account1@example.net): Got Axolotl signed prekey record:1
D/conversations( 4257): AxolotlService (account1@example.net): Got Axolotl signed prekey record:2
D/conversations( 4257): AxolotlService (account1@example.net): Building session for remote address: account2@example.net:1809345759
D/conversations( 4257): AxolotlService (account1@example.net): Building session for remote address: user@gmail.com:628754964
D/conversations( 4257): restoring messages
D/conversations( 4257): start merging phone contacts with roster
D/conversations( 4257): finished merging phone contacts
D/XmppDomainVerifier( 4257): searching for example.net in srvNames: [] xmppAddrs: [] domains:[example.net]
D/XmppDomainVerifier( 4257): searching for example.net in srvNames: [] xmppAddrs: [] domains:[example.net]
D/XmppDomainVerifier( 4257): domain example.net matched example.net
D/XmppDomainVerifier( 4257): domain example.net matched example.net
D/conversations( 4257): account2@example.net: TLS connection established
D/conversations( 4257): account1@example.net: TLS connection established
D/conversations( 4257): account2@example.net/phone: Authenticating with SCRAM-SHA-1
D/conversations( 4257): account1@example.net/phone: Authenticating with SCRAM-SHA-1
W/CursorWindow( 4257): Window is full: requested allocation 8077757 bytes, free space 2096464 bytes, window size 2097152 bytes
W/CursorWindow( 4257): Window is full: requested allocation 8077757 bytes, free space 2096464 bytes, window size 2097152 bytes
E/CursorWindow( 4257): Failed to read row 0, column 3 from a CursorWindow which has 0 rows, 15 columns.
E/AndroidRuntime( 4257): FATAL EXCEPTION: pool-3-thread-1
E/AndroidRuntime( 4257): Process: eu.siacs.conversations, PID: 4257
E/AndroidRuntime( 4257): java.lang.IllegalStateException: Couldn't read row 0, col 3 from CursorWindow.  Make sure the Cursor is initialized correctly before accessing data from it.
E/AndroidRuntime( 4257): 	at android.database.CursorWindow.nativeGetString(Native Method)
E/AndroidRuntime( 4257): 	at android.database.CursorWindow.getString(CursorWindow.java:438)
E/AndroidRuntime( 4257): 	at android.database.AbstractWindowedCursor.getString(AbstractWindowedCursor.java:51)
E/AndroidRuntime( 4257): 	at eu.siacs.conversations.entities.Message.fromCursor(Message.java:136)
E/AndroidRuntime( 4257): 	at eu.siacs.conversations.persistance.DatabaseBackend.getMessages(DatabaseBackend.java:410)
E/AndroidRuntime( 4257): 	at eu.siacs.conversations.persistance.DatabaseBackend.getMessages(DatabaseBackend.java:386)
E/AndroidRuntime( 4257): 	at eu.siacs.conversations.services.XmppConnectionService$18.run(XmppConnectionService.java:1101)
E/AndroidRuntime( 4257): 	at eu.siacs.conversations.utils.SerialSingleThreadExecutor$1.run(SerialSingleThreadExecutor.java:18)
E/AndroidRuntime( 4257): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
E/AndroidRuntime( 4257): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
E/AndroidRuntime( 4257): 	at java.lang.Thread.run(Thread.java:818)
D/ActivityManager( 1470): New dropbox entry: eu.siacs.conversations, data_app_crash, b5aca169-78e9-49d2-b055-5a8893ddf11b
I/CrashMonitor( 4192): CrashMonitorServiceBroadcastReceiver: Forwarding intent: android.intent.action.DROPBOX_ENTRY_ADDED tag: data_app_crash
I/CrashMonitor( 4192): CrashMonitorService: Service started...
I/CrashMonitor( 4192): CrashMonitorService: action: android.intent.action.DROPBOX_ENTRY_ADDED
````

hope this helps!

**Comment by axelsimon** [comment]
Any clues as to how to fix this and keep using Conversations without losing my current history of messages would be appreciated.

My understanding is that on startup, Conversations restores the messages, gets to the message with the picture and tries to put that picture into a temporary database, but as the picture is too big ("window is full…") it fails and then crashes.

Could removing the message referring to the picture in the sqlite db solve the problem?
I've tried moving the picture itself out of the way, but that hasn't helped.

Thanks!

axel

**Comment by philipflohr** [comment]
Hi,

can you try this branch: https://github.com/philipflohr/Conversations/tree/big_files_in_database
Please backup your application data as it will be deleted if you install a self-compiled version.
Restore the data after the installation and start Conversations.

**Comment by dchokola** [comment]
I am having the same issue. I'm using F-Droid, so here is alogcat output:
I/ActivityManager(28954): Process eu.siacs.conversations (pid 31464) (adj 1) has died.
I/WindowState(28954): WIN DEATH: Window{43461838 u0 eu.siacs.conversations/eu.siacs.conversations.ui.ConversationActivity}
V/SmartFaceService - 3rd party pause(28954): onReceive [android.intent.action.ACTIVITY_STATE/eu.siacs.conversations/create]
E/AndroidRuntime(31517): Process: eu.siacs.conversations, PID: 31517
E/AndroidRuntime(31517): 	at eu.siacs.conversations.persistance.FileBackend.resize(FileBackend.java:162)
E/AndroidRuntime(31517): 	at eu.siacs.conversations.persistance.FileBackend.getVideoPreview(FileBackend.java:414)
E/AndroidRuntime(31517): 	at eu.siacs.conversations.persistance.FileBackend.getThumbnail(FileBackend.java:386)
E/AndroidRuntime(31517): 	at eu.siacs.conversations.ui.XmppActivity$BitmapWorkerTask.doInBackground(XmppActivity.java:1288)
E/AndroidRuntime(31517): 	at eu.siacs.conversations.ui.XmppActivity$BitmapWorkerTask.doInBackground(XmppActivity.java:1273)
W/ActivityManager(28954):   Force finishing activity eu.siacs.conversations/.ui.ConversationActivity
I/WindowManager(28954): Screenshot max retries 4 of Token{43b09248 ActivityRecord{42715ad0 u0 eu.siacs.conversations/.ui.ConversationActivity t264 f}} appWin=Window{4369fae0 u0 Starting eu.siacs.conversations} drawState=4
V/SmartFaceService - 3rd party pause(28954): onReceive [android.intent.action.ACTIVITY_STATE/eu.siacs.conversations/pause]
I/WindowState(28954): WIN DEATH: Window{436ffab8 u0 eu.siacs.conversations/eu.siacs.conversations.ui.ConversationActivity}
I/ActivityManager(28954): Process eu.siacs.conversations (pid 31517) (adj 1) has died.
W/ActivityManager(28954): Service crashed 2 times, stopping: ServiceRecord{436a51e0 u0 eu.siacs.conversations/.services.XmppConnectionService}
I/ActivityManager(28954): Waited long enough for: ServiceRecord{44640c38 u0 eu.siacs.conversations/.services.XmppConnectionService}
V/SmartFaceService - 3rd party pause(28954): onReceive [android.intent.action.ACTIVITY_STATE/eu.siacs.conversations/create]
E/AndroidRuntime(32365): Process: eu.siacs.conversations, PID: 32365
E/AndroidRuntime(32365): 	at eu.siacs.conversations.persistance.FileBackend.resize(FileBackend.java:162)
E/AndroidRuntime(32365): 	at eu.siacs.conversations.persistance.FileBackend.getVideoPreview(FileBackend.java:414)
E/AndroidRuntime(32365): 	at eu.siacs.conversations.persistance.FileBackend.getThumbnail(FileBackend.java:386)
E/AndroidRuntime(32365): 	at eu.siacs.conversations.ui.XmppActivity$BitmapWorkerTask.doInBackground(XmppActivity.java:1288)
E/AndroidRuntime(32365): 	at eu.siacs.conversations.ui.XmppActivity$BitmapWorkerTask.doInBackground(XmppActivity.java:1273)
W/ActivityManager(28954): Process eu.siacs.conversations has crashed too many times: killing!
I/WindowManager(28954): Screenshot max retries 4 of Token{427e0eb8 ActivityRecord{427705f0 u0 eu.siacs.conversations/.ui.ConversationActivity t266 f}} appWin=Window{42715150 u0 Starting eu.siacs.conversations} drawState=4
I/ActivityManager(28954): Killing 32365:eu.siacs.conversations/u0a267 (adj 0): crash
I/WindowState(28954): WIN DEATH: Window{4275dd28 u0 eu.siacs.conversations/eu.siacs.conversations.ui.ConversationActivity}
V/SmartFaceService - 3rd party pause(28954): onReceive [android.intent.action.ACTIVITY_STATE/eu.siacs.conversations/pause]

**Comment by dchokola** [comment]
I tried to build the from the tree above but had an _INSANE_ time fighting with gradle and Android SDK. Does anyone have an apk already built?

**Comment by iNPUTmice** [comment]
@dchokola are you sure you are using the latest version? (1.13.7) this does look like something that has been fixed already.
